sql/core/src/main/scala/org/apache/spark/sql/execution/joins/JoinCodegenSupport.scala

```diff
@@ -42,13 +42,15 @@ trait JoinCodegenSupport extends CodegenSupport with BaseJoinExec {
       buildRow: Option[String] = None): (String, String, Seq[ExprCode]) = {
     val buildSideRow = buildRow.getOrElse(ctx.freshName("buildRow"))
     val buildVars = genOneSideJoinVars(ctx, buildSideRow, buildPlan, setDefaultValue = false)
+    val streamVars2 = streamVars.map(_.copy())
```

Comment:
late LGTM. Can we add some comments here? This is very tricky and we should tell others why a copy is needed. The problem is `ExprCode` is mutable, `streamVars` may be used again by full outer joins and we should not change its states here.